subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/recomp/RecompilationSpecProvider.java

```diff
@@ -52,10 +52,10 @@ private void processClasspathChanges(CurrentCompilation current, PreviousCompila
         for (Alignment<File> fileAlignment : alignment) {
             switch (fileAlignment.getKind()) {
                 case added:
-                    classpathEntryChangeProcessor.processChange(DefaultFileChange.added(fileAlignment.getCurrentValue().getAbsolutePath(), "classpathEntry", FileType.RegularFile), spec);
+                    classpathEntryChangeProcessor.processChange(DefaultFileChange.added(fileAlignment.getCurrentValue().getAbsolutePath(), "classpathEntry", FileType.RegularFile, ""), spec);
```

Comment:
Passing an empty string here seems arbitrary. We should have separate factory methods to create path-less changes, we could maybe even create a special type for them.

Reply:
It seems arbitrary, though it is not. These are classpath entry changes, so the normalized path of a classpath entry is always `""`.